core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -321,28 +320,29 @@ async def prune_ardupilot_processes(self) -> None:
                 raise ArdupilotProcessKillFail(f"Could not kill {process.name()}::{process.pid}.") from error
 
     async def kill_ardupilot(self) -> None:
-        self.should_be_running = False
-
-        if not self.current_board or self.current_board.platform != Platform.SITL:
-            try:
-                logger.info("Disarming vehicle.")
-                await self.vehicle_manager.disarm_vehicle()
-                logger.info("Vehicle disarmed.")
-            except Exception as error:
-                logger.warning(f"Could not disarm vehicle: {error}. Proceeding with kill.")
-
-        # TODO: Add shutdown command on HAL_SITL and HAL_LINUX, changing terminate/prune
-        # logic with a simple "self.vehicle_manager.shutdown_vehicle()"
-        logger.info("Terminating Ardupilot subprocess.")
-        await self.terminate_ardupilot_subprocess()
-        logger.info("Ardupilot subprocess terminated.")
-        logger.info("Pruning Ardupilot's system processes.")
-        await self.prune_ardupilot_processes()
-        logger.info("Ardupilot's system processes pruned.")
-
-        logger.info("Stopping Mavlink manager.")
-        self.mavlink_manager.stop()
-        logger.info("Mavlink manager stopped.")
+        try:
```

Comment:
Can't we just use try on the second block of code (the one with TODO), since this one has a try for a try ?